base_business_document_import/models/business_document_import.py

```diff
@@ -0,0 +1,871 @@
+            domain = [('supplier', '=', True)]
+            partner_type_label = _('supplier')
+        elif partner_type == 'customer':
+            domain = [('customer', '=', True)]
+            partner_type_label = _('customer')
+        else:
+            domain = []
+            partner_type_label = _('partner')
+        country = False
+        if partner_dict.get('country_code'):
+            countries = self.env['res.country'].search([
+                ('code', '=', partner_dict['country_code'])])
+            if countries:
+                country = countries[0]
+                domain += [
+                    '|',
+                    ('country_id', '=', False),
+                    ('country_id', '=', country.id)]
+            else:
+                chatter_msg.append(_(
+                    "The analysis of the business document returned '%s' as "
+                    "country code. But there are no country with that code "
+                    "in Odoo.") % partner_dict['country_code'])
+        if country and partner_dict.get('state_code'):
+            states = self.env['res.country.state'].search([
+                ('code', '=', partner_dict['state_code']),
+                ('country_id', '=', country.id)])
+            if states:
+                domain += [
+                    '|',
+                    ('state_id', '=', False),
+                    ('state_id', '=', states[0].id)]
+        # Hook to plug alternative matching methods
+        partner = self._hook_match_partner(
+            partner_dict, chatter_msg, domain, partner_type_label)
+        if partner:
+            return partner
+        if partner_dict.get('vat'):
+            vat = partner_dict['vat'].replace(' ', '').upper()
+            # use base_vat_sanitized
+            partners = rpo.search(
+                domain + [
+                    ('parent_id', '=', False),
+                    ('sanitized_vat', '=', vat)])
+            if partners:
+                return partners[0]
+            else:
+                chatter_msg.append(_(
+                    "The analysis of the business document returned '%s' as "
+                    "%s VAT number. But there are no %s "
+                    "with this VAT number in Odoo.")
+                    % (vat, partner_type_label, partner_type_label))
+        website_domain = False
+        email_domain = False
+        if partner_dict.get('email') and '@' in partner_dict['email']:
+            partners = rpo.search(
+                domain + [('email', '=ilike', partner_dict['email'])])
+            if partners:
+                return partners[0]
+            else:
+                email_domain = partner_dict['email'].split('@')[1]
+        if partner_dict.get('website'):
+            urlp = urlparse(partner_dict['website'])
+            netloc = urlp.netloc
+            if not urlp.scheme and not netloc:
+                netloc = urlp.path
+            if netloc and netloc.split('.') >= 2:
```

Comment:
I think it should be:

```
if netloc and len(netloc.split('.')) >= 2:
```

otherwise I get this error:

```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/home/onestein/Pycharm/Gitlab/odoo-11.0/src/odoo-11.0/odoo/http.py", line 646, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/onestein/Pycharm/Gitlab/odoo-11.0/src/odoo-11.0/odoo/http.py", line 307, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/home/onestein/Pycharm/Gitlab/odoo-11.0/src/odoo-11.0/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/home/onestein/Pycharm/Gitlab/odoo-11.0/src/odoo-11.0/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/home/onestein/Pycharm/Gitlab/odoo-11.0/src/odoo-11.0/odoo/http.py", line 339, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/onestein/Pycharm/Gitlab/odoo-11.0/src/odoo-11.0/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/onestein/Pycharm/Gitlab/odoo-11.0/src/odoo-11.0/odoo/http.py", line 332, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/onestein/Pycharm/Gitlab/odoo-11.0/src/odoo-11.0/odoo/http.py", line 927, in __call__
    return self.method(*args, **kw)
  File "/home/onestein/Pycharm/Gitlab/odoo-11.0/src/odoo-11.0/odoo/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/home/onestein/Pycharm/Gitlab/odoo-11.0/src/odoo-11.0/addons/web/controllers/main.py", line 934, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/onestein/Pycharm/Gitlab/odoo-11.0/src/odoo-11.0/addons/web/controllers/main.py", line 922, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/onestein/Pycharm/Gitlab/odoo-11.0/src/odoo-11.0/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/home/onestein/Pycharm/Gitlab/odoo-11.0/src/odoo-11.0/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/onestein/Pycharm/Gitlab/odoo-11.0/src/oca/account_invoice_import/wizard/account_invoice_import.py", line 401, in import_invoice
    parsed_inv['partner'], parsed_inv['chatter_msg'])
  File "/home/onestein/Pycharm/Gitlab/odoo-11.0/src/oca/base_business_document_import/models/business_document_import.py", line 134, in _match_partner
    if netloc and netloc.split('.') >= 2:
TypeError: unorderable types: list() >= int()
```